specification/devcenter/data-plane/Microsoft.DevCenter/stable/2023-04-01/devbox.json

```diff
@@ -0,0 +1,1825 @@
+          {
+            "$ref": "devcenter.json#/parameters/ProjectNameParameter"
+          },
+          {
+            "$ref": "devcenter.json#/parameters/UserIdParameter"
+          },
+          {
+            "$ref": "#/parameters/DevBoxNameParameter"
+          },
+          {
+            "name": "body",
+            "in": "body",
+            "description": "Represents a environment.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DevBox"
+            }
+          }
+        ],
+        "description": "Creates or replaces a Dev Box.",
+        "operationId": "DevBoxes_CreateDevBox",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "original-uri"
+        },
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DevBox"
+            }
+          },
+          "201": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DevBox"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "devcenter.json#/definitions/CloudError"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "description": "The error code for specific error that occurred.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DevBoxes_CreateDevBox": {
+            "$ref": "./examples/DevBoxes_Create.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Dev Boxes"
+        ],
+        "parameters": [
+          {
+            "$ref": "devcenter.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "devcenter.json#/parameters/ProjectNameParameter"
+          },
+          {
+            "$ref": "devcenter.json#/parameters/UserIdParameter"
+          },
+          {
+            "$ref": "#/parameters/DevBoxNameParameter"
+          }
+        ],
+        "description": "Deletes a Dev Box.",
+        "operationId": "DevBoxes_DeleteDevBox",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "operation-location"
+        },
+        "responses": {
+          "202": {
+            "description": "The request was accepted.",
+            "schema": {
+              "$ref": "devcenter.json#/definitions/OperationStatus"
+            },
+            "headers": {
+              "Operation-Location": {
+                "description": "URL to query for status of the operation.",
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "Resource does not exist."
+          },
```

Comment:
This looks like something we missed in our previous reviews. Long-running operations (that return 202) should not also have synchronous success responses because it complicates the client logic. Does your service really work this way? If so, is it too late to fix this?

Reply:
@mikekistler Thanks for pointing this out, we were not aware this was an anti-pattern. This is indeed how the service acts in practice and what we've built our SDKs/clients around (delete on non-existent resource will return 204 rather than returning an already-complete LRO). Our control plane spec has the same behavior for async deletes, as does our sister service (DTL) which I believe is why we thought this was standard practice. 

Do you know what kind of problems have come up with this pattern in the past? I don't think we've run into any issues in the SDK/CLI/client consumption due to this so far, but it's possible we're missing a scenario.

Reply:
We don't have concrete cases where this caused problems. Our SDKs handle this because there are other services that do this, so the problems would happen in code that customers write to call the REST API directly. The problem is exacerbated by the fact that our SDK generators often use the 200 or 204 response to describe the _eventual_ result of an LRO, as opposed to a response actually returned from the initiating request.

But this one is on me because I missed it in the earlier reviews. If your service works this way and you are on the cusp of GA it would be wrong to hold you up for this.

Reply:
Understood, thanks! We'll keep this in mind for future APIs then.